### PR TITLE
More exact computation of allocations in Sparse broadcast

### PR DESCRIPTION
**Pull request by mbauman**
~~DO NOT MERGE YET. This includes some debugging code to run this test 10000 times on each CI. It passed locally (on a Linux box) with over 100000 runs.~~

This uses `@allocated` to actually compute how expensive it is to convert a transposed sparse matrix to a CSC structure instead of just guessing.

### CODE REVIEW

File: stdlib/SparseArrays/test/higherorderfns.jl [hunk 1]
```diff
@@ -310,6 +310,7 @@ end
 end
 
 @testset "broadcast[!] over combinations of scalars and sparse vectors/matrices" begin
+for _ in 1:10000
```
Comment (Sacha0) [inline stdlib/SparseArrays/test/higherorderfns.jl]: 😂 Some lucky CI nodes are going to have a fun afternoon.

**Comment by mbauman** [comment]
macOS failures were in FileWatching and Random

```
Error in testset FileWatching:
Test Failed at /private/tmp/julia/share/julia/stdlib/v0.7/FileWatching/test/runtests.jl:401
  Expression: pop!(changes) == (F_PATH => FileWatching.FileEvent(FileWatching.UV_RENAME))
   Evaluated: "afile.txt3" => FileWatching.FileEvent(true, false, false) == "afile.txt" => FileWatching.FileEvent(true, false, false)
# ...
Error in testset Random:
Test Failed at /private/tmp/julia/share/julia/stdlib/v0.7/Random/test/runtests.jl:123
  Expression: rand(s) != rand(s)
   Evaluated: 170141183460469231731687303715884107957 != 170141183460469231731687303715884107957
ERROR: LoadError: Test run finished with errors
in expression starting at /private/tmp/julia/share/julia/test/runtests.jl:59
```

win64 failures were in loading:

```
Error in testset loading:
Test Failed at C:\projects\julia\julia-8a00f32ed2\share\julia\test\loading.jl:474
  Expression: path == get(paths, id, nothing)
   Evaluated: "C:\\Users\\appveyor\\AppData\\Local\\Temp\\1\\jl_751D.tmp\\packages\\C\\OmQA\\src\\C.jl" == "C:\\Users\\appveyor\\AppData\\Local\\Temp\\1\\jl_2FD2.tmp\\packages\\C\\OmQA\\src\\C.jl"
# …
```

Looks like this worked.

**Comment by KristofferC** [comment]
cc @StefanKarpinski for loading failure.

File: stdlib/SparseArrays/test/higherorderfns.jl [hunk 2]
```diff
@@ -356,29 +356,29 @@ end
...
+     @test broadcast(*, sparseargs...) == sparse(broadcast(*, denseargs...))
+     @test isa(@inferred(broadcast(*, sparseargs...)), SparseMatrixCSC{elT})
+     # test broadcast! entry point
+     fX = broadcast(*, sparseargs...); X = sparse(fX)
+     @test broadcast!(*, X, sparseargs...) == sparse(broadcast!(*, fX, denseargs...))
+     @test isa(@inferred(broadcast!(*, X, sparseargs...)), SparseMatrixCSC{elT})
+     X = sparse(fX) # reset / warmup for @allocated test
+     @test_broken (@allocated broadcast!(*, X, sparseargs...)) == 0
+     X = sparse(fX) # reset / warmup for @allocated test
+     @test (@allocated broadcast!(*, X, sparseargs...)) <= 128
```
Comment (Sacha0) [inline stdlib/SparseArrays/test/higherorderfns.jl]: Indentation mishap? :)

Comment (mbauman) [reply stdlib/SparseArrays/test/higherorderfns.jl]: Whoops, thanks!

**Review by Sacha0 — APPROVED** [review]
lgtm! :)

**Comment by mbauman** [comment]
FreeBSD timed out ("command timed out: 1200 seconds without output running …")

**Pull request merged**